Assets/MixedRealityToolkit/UX/Scripts/Dialog/DialogShell.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+
+using System;
+using System.Text;
+using UnityEngine;
+using MixedRealityToolkit.Utilities.Solvers;
+using MixedRealityToolkit.UX.Buttons;
+
+#if UNITY_WSA && UNITY_2017_2_OR_NEWER
+using System.Collections.Generic;
+using UnityEngine.XR;
+using UnityEngine.XR.WSA;
+#endif
+
+
+namespace MixedRealityToolkit.UX.Dialog
+{
+    /// <summary>
+    /// Dialog that approximates the look of a Hololens shell dialog
+    /// </summary>
+    public class DialogShell : Dialog
+    {
+        public int MaxCharsPerLine = 60;
+
+        [SerializeField]
+        private TextMesh titleText;
+
+        [SerializeField]
+        private TextMesh messageText;
+
+        [SerializeField]
+        private GameObject[] oneButtonSet;
+
+        [SerializeField]
+        private GameObject[] twoButtonSet;
+
+        private DialogButton buttonPressed;
+
+        protected void OnDrawGizmos()
+        {
+            if (messageText != null)
+            {
+                messageText.text = WordWrap(messageText.text, MaxCharsPerLine);
+            }
+        }
+
+        protected override void FinalizeLayout()
+        {
+            // Optimize the content for immersive headset
+            if (HolographicSettings.IsDisplayOpaque)
```

Comment:
we're not wrapping this in a compiler symbol so this would not build for Standanlone as is.